server/config/config.go

```diff
@@ -230,6 +231,20 @@ type (
 	}
 )
 
+// State Streaming configuration
+type (
+	// StreamingConfig defines application configuration for external streaming services
+	StreamingConfig struct {
+		ABCI ABCIListenerConfig `mapstructure:"abci"`
+	}
+	// ABCIListenerConfig defines application configuration for ABCIListener streaming service
+	ABCIListenerConfig struct {
+		Keys          []string `mapstructure:"keys"`
+		Plugin        string   `mapstructure:"plugin"`
+		StopNodeOnErr bool     `mapstructure:"stop-node-on-err"`
```

Comment:
I can't find where this is being enforced. I see in `registerABCIListenerPlugin` that it's being read and provided to the `StreamingManager`, but the the manager doesn't use it at all. Am I missing something?

Reply:
They're used when the config file is first loaded when the server starts up and when it's generated. They ensure that the config is correct when it's Unmarshalled. Beyond that, there are not any references for other properties other than the server startup. Since this is part of the `/server/config/` package it shouldn't be mixed in with the App API. At least that's what I concluded when I couldn't find other uses outside of the server startup process. 

https://github.com/provenance-io/cosmos-sdk/blob/9777eb95323564735eb209acf52fd84b94d20547/server/config/config.go#L367-L373

Reply:
Ah, I see it now. Yes that functionality has moved to the plugin and it's my mistake that it didn't make it in during the cherry-pick. fixing it.

Reply:
compilation errors fixed.